### PR TITLE
(spyglass/lenses) allow configuration sandbox permissions

### PR DESCRIPTION
**Pull request by norrs**
This provides the ability to configure iframe sandbox permissions pr lense. This allows the operator of the prow installation to define which permissions it trust to each lense.

PR comes from the ideas and discussions in
https://github.com/kubernetes-sigs/prow/pull/294

### CODE REVIEW

File: cmd/checkconfig/testdata/combined.yaml [hunk 1]
```diff
@@ -23,6 +23,10 @@ deck:
           - (FAIL|Failure \[)\b
           - panic\b
           - ^E\d{4} \d\d:\d\d:\d\d\.\d\d\d]
+          iframe_sandbox_permissions:
+          - allow-scripts
+          - allow-popups
+          - allow-popups-to-escape-sandbox
```
Comment (norrs) [inline cmd/checkconfig/testdata/combined.yaml:29]: Updated this as it looked like a sane idea to do if this is supposed to demostrate a full configuration file example.

File: pkg/spyglass/lenses/buildlog/lens.go [hunk 2]
```diff
@@ -97,6 +99,17 @@ func (lens Lens) Header(artifacts []api.Artifact, resourceDir string, config jso
...
+// defaultSandboxPermissions is the default value for iframe_sandbox_permissions lense config if it is not specified.
+var defaultSandboxPermissions = strings.Join(
+	[]string{
+		"allow-scripts",
+		"allow-top-navigation",
+		"allow-popups",
+		"allow-same-origin",
+	},
+	" ",
+)
```
Comment (norrs) [inline pkg/spyglass/lenses/buildlog/lens.go:111]: This is defaultSandboxPermissions for the parsedConfig, tho I personally like to work with arrays instead of strings with `foo bar zar`. 

Not sure if you prefer this or the `allow-scripts allow-top-navigation allow-popups allow-same-origin` instead. 

Please let me know, I'll update if you want to change it.

Comment (norrs) [reply pkg/spyglass/lenses/buildlog/lens.go:111]: Suggestion is to do another PR after this to adjust the defaults to not combine `allow-same-origin` together with `allow-scripts`, but I suppose this needs to be communicated to the users of the prow installation, so they are aware of the ability to configure this pr lense.

**Comment by michelle192837** [comment]
Thanks for the ping and apologies for the delay, this dropped off my radar >< 

I'll review today!

File: pkg/spyglass/lenses/buildlog/lens_test.go [hunk 3]
```diff
@@ -61,6 +62,14 @@ func TestGetConfig(t *testing.T) {
 				}
 				return d
 			}(),
+		}, {
```
Comment (michelle192837) [inline pkg/spyglass/lenses/buildlog/lens_test.go:65]: nit: Add test case verifying what the iframe permissions are if none are specified.

**Review by michelle192837 — COMMENTED** [review]
LGTM aside from a minor nit.

File: pkg/spyglass/lenses/buildlog/lens_test.go (hunk 3 above)
Comment (norrs) [reply pkg/spyglass/lenses/buildlog/lens_test.go:65]: Added test @michelle192837 , good nit 👍

**Comment by norrs** [comment]
Seems like we need an approval from [cmd/checkconfig/OWNERS](https://github.com/kubernetes-sigs/prow/blob/main/cmd/checkconfig/OWNERS): @chases2 , @stevekuznetsov or @cjwagner

**Comment by michelle192837** [comment]
Those folks are no longer working on Prow unfortunately. I think this should get approval from someone in https://github.com/kubernetes-sigs/prow/blob/main/OWNERS until that OWNERS file is fixed.

(Pulling randomly) @petr-muller  or @matthyx ?

**Review by matthyx — APPROVED** [review]
/lgtm
